File: scripts/backup.py
import firebase_admin
from firebase_admin import credentials
from firebase_admin import db
import json
import datetime
import os
from dotenv import load_dotenv

load_dotenv()

# Set up Firebase Admin SDK
cred = credentials.Certificate('../service_account.json') # replace with your own service account key
firebase_admin.initialize_app(cred, {
    'databaseURL': os.getenv("DATABASE_URL") # replace with your own database URL
})

# Backup the entire database to a JSON file
ref = db.reference('/')
data = ref.get()
filename = 'backup.json'
with open(filename, 'w') as outfile: # replace with your desired output filename
    json.dump(data, outfile)

File: scripts/backup.js
const admin = require("firebase-admin");
const fs = require("fs");

const serviceAccount = require("../service_account.json");

admin.initializeApp({
  credential: admin.credential.cert(serviceAccount),
  databaseURL: "https://tennis-app-7dabe.firebaseio.com"
});

const database = admin.database();

const backupRef = database.ref();

backupRef.once("value", (snapshot) => {
  const backupData = JSON.stringify(snapshot.val(), null, 2);
  fs.writeFileSync("./backup.json", backupData);
  console.log("Backup created successfully.");
  process.exit(0)
});

